ambry-replication/src/main/java/com.github.ambry.replication/DiskTokenPersistor.java

```diff
@@ -62,6 +67,10 @@ protected void persist(String mountPath, List<ReplicaTokenInfo> tokenInfoList) t
       logger.debug("Completed writing replica tokens to file {}", actual.getAbsolutePath());
     } catch (IOException e) {
       logger.error("IO error while persisting tokens to disk {}", temp.getAbsoluteFile());
+      // check disk state in storageManager. If there is a hardware issue, persistor should skip this bad disk next time.
+      if (!storageManager.isDiskAvailableAtMountPath(mountPath)) {
```

Comment:
You don't need to add a new method to StorageManager to check by mount path.  This class already has the partitionGroupedByMountPath map that you can use to find the partition/replica for the mount path.  Then you can call StoreManager.checkLocalPartitionStatus() and avoid adding replication -> store dependency.

Reply:
Good point, will refactor as you suggested